File: packages/urdu-js/src/analysis/rtl.ts
export function isRTL(text: string): boolean {
  for (const ch of text) {
    const cp = ch.codePointAt(0) ?? 0
    if ((cp >= 0x0590 && cp <= 0x08ff) || (cp >= 0xfb1d && cp <= 0xfdff) || (cp >= 0xfe70 && cp <= 0xfeff)) return true
    if ((cp >= 0x0041 && cp <= 0x005a) || (cp >= 0x0061 && cp <= 0x007a)) return false
  }
  return false
}

File: packages/urdu-dart/lib/src/analysis/rtl.dart
/// Returns true if the first strong directional character is RTL.
bool isRTL(String text) {
  for (final r in text.runes) {
    if ((r >= 0x0590 && r <= 0x08ff) ||
        (r >= 0xfb1d && r <= 0xfdff) ||
        (r >= 0xfe70 && r <= 0xfeff)) {
      return true;
    }
    if ((r >= 0x0041 && r <= 0x005a) || (r >= 0x0061 && r <= 0x007a)) {
      return false;
    }
  }
  return false;
}
